### ISSUE TITLE
In Ubuntu 20.10

### ISSUE THREAD

**Issue by yegnal** [flathub/org.gimp.GIMP.Plugin.Resynthesizer#1]
Traceback (most recent call last):
File “/app/lib/gimp/2.0/python/gimpfu.py”, line 740, in response
dialog.res = run_script(params)
File “/app/lib/gimp/2.0/python/gimpfu.py”, line 361, in run_script
return apply(function, params)
File “/home/me/Documents/GIMP/Plugins/plugin-heal-selection.py”, line 148, in heal_selection
pdb.plug_in_resynthesizer(timg, tdrawable, 0,0, useBorder, work_drawable.ID, -1, -1, 0.0, 0.117, 16, 500)
error: procedure not found

**Comment by hfiguiere** [comment]
If you could describe what you are doing that would be helpful.

**Comment by yegnal** [comment]
Using flatpak version of gimp, and trying to use the resynthesizer plugin in Ubuntu 20.10



    On Wednesday, February 3, 2021, 08:24:58 AM EST, Hubert Figuière <notifications@github.com> wrote:  
 
 


If you could describe what you are doing that would be helpful.

**Comment by bootchk** [comment]
Did you install the flatpaked resynthesizer, or from some other place?  Or could an old version of resynthesizer be laying around where the flatpaked GIMP finds it.  If you install the flatpak GIMP, you should install the flatpak resynthesizer.  Off the top of my head I can't remember the flatpak command.

From the GIMP point of view, this is the PDB (Procedural Data Base) portion of GIMP saying that the procedure (in this case named 'plug-in-resynthesizer' (where the name is the name registered in the PDB) could not be found.  Procedures are registered in the PDB at GIMP start time.  In this case, it means the the file for the procedure is probably not in the right place, or the search paths that GIMP uses at startup time are not correct.  The file should be an executable file having name 'plug_in_resynthesizer' ( as I recall, the hyphenation hard to remember.).  And it should be installed either to the same place as GIMP installs its own plugins ( without flatpak, something like /usr/share/ ..../ GIMP/2.0/plug-ins) or to the user specific, custom install location (without flatpak ~/.config/GIMP/2.0/plug-ins.)  With flatpak, the paths are different.

**Comment by hfiguiere** [comment]
Sill not a lot of details though. 

Simple question: did you install the Resynthesizer flatpak as well (it's on flathub) ?

`flatpak install org.gimp.GIMP.Plugin.Resynthesizer` in the terminal should do the trick

**Comment by yegnal** [comment]
This is flatpak list:    Name                                               Application ID                                 Version           Branch           Installationdefault                                                org.freedesktop.Platform.GL.default                              19.08            systemopenh264                                           org.freedesktop.Platform.openh264              2.1.0             2.0              systemGNU Image Manipulation Program   org.gimp.GIMP                                  2.10.22           stable           systemManual                                               org.gimp.GIMP.Manual                                             2.10             systemBIMP                                                  org.gimp.GIMP.Plugin.BIMP                      2.5               2-3.36           systemFocusBlur                                          org.gimp.GIMP.Plugin.FocusBlur                 3.2.6             2-3.36           systemFourier                                               org.gimp.GIMP.Plugin.Fourier                   0.4.3             2-3.36           systemG'MIC                                                org.gimp.GIMP.Plugin.GMic                      2.9.3             2-3.36           systemGimpLensfun                                     org.gimp.GIMP.Plugin.Lensfun                   0.2.4             2-3.36           systemLiquidRescale                                    org.gimp.GIMP.Plugin.LiquidRescale             0.7.2             2-3.36           systemResynthesizer                                   org.gimp.GIMP.Plugin.Resynthesizer             2.0.3             2-3.36           systemGNOME Application Platform version 3.36            org.gnome.Platform                                               3.36             systemYaru Gtk Theme                                     org.gtk.Gtk3theme.Yaru                                           3.22             system


So when I ran Gimp, the heal tool did not appear in Gimp so I added a path to where I have the resysnthesizer .py scripts and then it appears in the filters/enhance menu, but it doesn't work with the error I provided.  
Again, without setting the path on my own to the folder where the .py resynthesizer plugin is it doesn't even appear in the menu as an option and if I do it fails.    On Wednesday, February 3, 2021, 01:53:35 PM EST, Yegnal <yegnal@yahoo.com> wrote:  
 
  I installed the flatpak resynthesiser after purging the apt version


    On Wednesday, February 3, 2021, 12:44:56 PM EST, Hubert Figuière <notifications@github.com> wrote:  
 
 


Sill not a lot of details though.

Simple question: did you install the Resynthesizer flatpak as well (it's on flathub) ?

flatpak install org.gimp.GIMP.Plugin.Resynthesizer in the terminal should do the trick

**Comment by yegnal** [comment]
I installed the flatpak resynthesizer after purging the apt version


    On Wednesday, February 3, 2021, 11:53:23 AM EST, Lloyd Konneker <notifications@github.com> wrote:  
 
 


Did you install the flatpaked resynthesizer, or from some other place? Or could an old version of resynthesizer be laying around where the flatpaked GIMP finds it. If you install the flatpak GIMP, you should install the flatpak resynthesizer. Off the top of my head I can't remember the flatpak command.

From the GIMP point of view, this is the PDB (Procedural Data Base) portion of GIMP saying that the procedure (in this case named 'plug-in-resynthesizer' (where the name is the name registered in the PDB) could not be found. Procedures are registered in the PDB at GIMP start time. In this case, it means the the file for the procedure is probably not in the right place, or the search paths that GIMP uses at startup time are not correct. The file should be an executable file having name 'plug_in_resynthesizer' ( as I recall, the hyphenation hard to remember.). And it should be installed either to the same place as GIMP installs its own plugins ( without flatpak, something like /usr/share/ ..../ GIMP/2.0/plug-ins) or to the user specific, custom install location (without flatpak ~/.config/GIMP/2.0/plug-ins.) With flatpak, the paths are different.

**Comment by yegnal** [comment]
I installed the flatpak resynthesiser after purging the apt version


    On Wednesday, February 3, 2021, 12:44:56 PM EST, Hubert Figuière <notifications@github.com> wrote:  
 
 


Sill not a lot of details though.

Simple question: did you install the Resynthesizer flatpak as well (it's on flathub) ?

flatpak install org.gimp.GIMP.Plugin.Resynthesizer in the terminal should do the trick

**Comment by yegnal** [comment]
When I do a search as root I find the flatpak resynthesizer plugins at:/var/lib/flatpak/runtime/org.gimp.GIMP.Plugin.Resynthesizer/x86_64/2-3.36/ece209307f93889a3b75c31a809bd87b318ea42f014473f21b0a4604643c0b69/files/plug-ins

That directory does not appear inside of flatpak Gimp folder preferences, adding it in Gimp preferences as a folder location for plugins doesn't help and actually the resynthesizer don't even appear in the filters menu.
The only way it seems I can have them show up in the Gimp filters menu is by adding my own location for the plugins in my home directory, they just don't work giving that error.  They are executable....
    On Wednesday, February 3, 2021, 02:03:19 PM EST, Yegnal <yegnal@yahoo.com> wrote:  
 
  This is flatpak list:    Name                                               Application ID                                 Version           Branch           Installationdefault                                                org.freedesktop.Platform.GL.default                              19.08            systemopenh264                                           org.freedesktop.Platform.openh264              2.1.0             2.0              systemGNU Image Manipulation Program   org.gimp.GIMP                                  2.10.22           stable           systemManual                                               org.gimp.GIMP.Manual                                             2.10             systemBIMP                                                  org.gimp.GIMP.Plugin.BIMP                      2.5               2-3.36           systemFocusBlur                                          org.gimp.GIMP.Plugin.FocusBlur                 3.2.6             2-3.36           systemFourier                                               org.gimp.GIMP.Plugin.Fourier                   0.4.3             2-3.36           systemG'MIC                                                org.gimp.GIMP.Plugin.GMic                      2.9.3             2-3.36           systemGimpLensfun                                     org.gimp.GIMP.Plugin.Lensfun                   0.2.4             2-3.36           systemLiquidRescale                                    org.gimp.GIMP.Plugin.LiquidRescale             0.7.2             2-3.36           systemResynthesizer                                   org.gimp.GIMP.Plugin.Resynthesizer             2.0.3             2-3.36           systemGNOME Application Platform version 3.36            org.gnome.Platform                                               3.36             systemYaru Gtk Theme                                     org.gtk.Gtk3theme.Yaru                                           3.22             system


So when I ran Gimp, the heal tool did not appear in Gimp so I added a path to where I have the resysnthesizer .py scripts and then it appears in the filters/enhance menu, but it doesn't work with the error I provided.  
Again, without setting the path on my own to the folder where the .py resynthesizer plugin is it doesn't even appear in the menu as an option and if I do it fails.    On Wednesday, February 3, 2021, 01:53:35 PM EST, Yegnal <yegnal@yahoo.com> wrote:  
 
  I installed the flatpak resynthesiser after purging the apt version


    On Wednesday, February 3, 2021, 12:44:56 PM EST, Hubert Figuière <notifications@github.com> wrote:  
 
 


Sill not a lot of details though.

Simple question: did you install the Resynthesizer flatpak as well (it's on flathub) ?

flatpak install org.gimp.GIMP.Plugin.Resynthesizer in the terminal should do the trick

**Comment by hfiguiere** [comment]
Not sure why it is looking at the plugin in `“/home/me/Documents/GIMP/Plugins/plugin-heal-selection.py”` but it shouldn't be, and that's probably what the problem is.

**Comment by hfiguiere** [comment]
`~/Documents/GIMP/Plugins` isn't in the default plugin path either.

**Comment by yegnal** [comment]
I have installed flatpak Gimp and flatpak resysnthesizer.  

flatpak list:
GNU Image Manipulation Program              org.gimp.GIMP                             2.10.24       stable       system
Resynthesizer                               org.gimp.GIMP.Plugin.Resynthesizer             2.0.3           2-40         system

The heal selection tool does NOT appear in the menu.  In Gimp preferences, the folder location for plugins are:
/home/me/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins
/app/lib/gimp/2.0/plug-ins

The heal selection tool does NOT appear in the Gimp menu.

I can make the heal selection tool appear in the Gimp editing menu by adding a folder location in Gimp preferences to the heal selection scripts I previously used in the apt version of Gimp, but I can't make them work.  If in fact I do just that, I get errors:

Traceback (most recent call last):
  File "/app/lib/gimp/2.0/python/gimpfu.py", line 740, in response
    dialog.res = run_script(params)
  File "/app/lib/gimp/2.0/python/gimpfu.py", line 361, in run_script
    return apply(function, params)
  File "/home/me/Documents/GIMP/Plugins/plugin-heal-selection.py", line 148, in heal_selection
    pdb.plug_in_resynthesizer(timg, tdrawable, 0,0, useBorder, work_drawable.ID, -1, -1, 0.0, 0.117, 16, 500)
error: procedure not found


This remains the current condition even after a recent update to flatpak Gimp, currently 2.10.24

**Comment by yegnal** [comment]
So after revisiting this just now, unfortunately after my previous post a minute ago, I took another look..

I removed flatpak Gimp and plugins;

I removed some directories I found on my system with:   find -H -maxdepth 3 -iname "gimp"
Then removed:
sudo rm -r ./snap/gimp
sudo rm -r ./.config/GIMP/
sudo rm -r ./.cache/gimp/ 

I reinstalled flatpak Gimp and plugins, heal selection works now.

The one thing I notice is that there now exists a new folder location in Gimp preferences under plugin folder locations that did NOT exist before I did the above.

/app/extensions/plug-ins

This was NOT there before I did what I outlined above, and now appears at the bottom of the plugin folder location list in Gimp preferences, seems to be key to the successful running of the heal selection script.

I wonder why it didn't appear before ?
